lib/customer.rb

```diff
@@ -0,0 +1,36 @@
+require 'csv'
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address
+
+  @@all_customers = nil
+
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all
+    unless @@all_customers
+      @@all_customers = CSV.read("data/customers.csv")
+                            .map {|row| Customer.new(row[0].to_i,
+                                                     row[1],
+                                                     {street: row[2], city: row[3], state: row[4], zip: row[5]})}
+    end
+    return @@all_customers
```

Comment:
2 things! This indentation leaves something to be desired I think. I would rather this be less horizontally spread out, a single indent on map would do to make it clear that `.map` is part of the above code.

Second, I see that you want to avoid running this multiple times, but having a class variable isn't going to be a solution I want you to get too cozy with. The better thing to do here would be to use a composition relationship that has a single third class manage both `Customer`s and `Order`s.